templates/http/template/connector.go

```diff
@@ -0,0 +1,27 @@
+package template
+
+import (
+	gohttp "net/http"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/log"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+)
+
+// Connector injects an HTTP request with AWS authorization headers.
+type Connector struct {
+	logger log.Logger
+}
+
+// This function has access to the client http.Request and the credentials (as a map), and is expected to decorate the
+// request with Authorization headers.
```

Comment:
line too long: wrap at 80 (you can set a hotkey in GoLand to make this easier)

Also, I'd say:

> and is expected to modify the request so that it will authenticate.  This typically means adding required authorization headers.

because some auth systems require other types of modifications, and in theory there is no limit on what they might require.